clients/v1_auth/v1_auth.go

```diff
@@ -0,0 +1,428 @@
+package v1_auth
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/influxdata/influx-cli/v2/api"
+	"github.com/influxdata/influx-cli/v2/clients"
+	"github.com/influxdata/influx-cli/v2/pkg/influxid"
+)
+
+type Client struct {
+	clients.CLI
+	api.AuthorizationsApi
+	api.UsersApi
+	api.OrganizationsApi
+}
+
+type v1PrintOpts struct {
+	deleted bool
+	token   *v1Token
+	tokens  []v1Token
+}
+
+type v1Token struct {
+	ID          string   `json:"id"`
+	Description string   `json:"description"`
+	Token       string   `json:"token"`
+	Status      string   `json:"status"`
+	UserName    string   `json:"userName"`
+	UserID      string   `json:"userID"`
+	Permissions []string `json:"permissions"`
+}
+
+type CreateParams struct {
+	clients.OrgParams
+	Desc        string
+	Username    string
+	Password    string
+	NoPassword  bool
+	ReadBucket  []string
+	WriteBucket []string
+}
+
+func (c Client) Create(ctx context.Context, params *CreateParams) error {
+	if params.Password != "" && params.NoPassword {
+		return fmt.Errorf("only one of --password and --no-password may be specified")
+	}
+
+	orgID, err := c.getOrgID(ctx, params.OrgParams)
+	if err != nil {
+		return err
+	}
+
+	ids, err := c.UsersApi.GetUsers(ctx).Name(params.Username).Execute()
+	if err != nil || len(ids.GetUsers()) == 0 {
+		return fmt.Errorf("could not get user from Username %q: %w", params.Username, err)
+	}
+
+	// verify an existing token with the same username doesn't already exist
+	auth, err := c.AuthorizationsApi.GetAuthorizationsID(ctx, ids.GetUsers()[0].GetId()).Execute()
+	authPtr := &auth
+	if &authPtr != nil {
+		return fmt.Errorf("authorization with username %q already exists", params.Username)
+	}
+	if err != nil {
+		return fmt.Errorf("failed to verify username %q has no auth: %w", params.Username, err)
+	}
+
+	password := params.Password
+	if password == "" && !params.NoPassword {
+		password, err = c.CLI.StdIO.GetSecret("Please enter your password", 0)
+		if err != nil {
+			return err
+		}
+	}
+
+	bucketPerms := []struct {
+		action string
+		perms  []string
+	}{
+		{action: "read", perms: params.ReadBucket},
+		{action: "write", perms: params.WriteBucket},
+	}
+
+	var permissions []api.Permission
+	for _, bp := range bucketPerms {
+		for _, p := range bp.perms {
+			// verify the input ID
+			if _, err := influxid.IDFromString(p); err != nil {
+				return fmt.Errorf("invalid bucket ID '%s': %w (did you pass a bucket name instead of an ID?)", p, err)
+			}
+
+			newPerm := api.Permission{
+				Action: bp.action,
+				Resource: api.PermissionResource{
+					Type:  "buckets",
+					Id:    *api.NewNullableString(&p),
```

Comment:
The `*api.NewNullableString(...)` wrapper feels unnecessary here... can we use `p` and `orgID` directly?

Reply:
We would have to update the yaml's to allow for it since it requires this type currently